checker_test.go

```diff
@@ -2156,8 +2168,203 @@ error:
 error:
   bad check: at index 0: bad check: regexp is not a string
 `,
+}, {
+	about:   "JSONEquals simple",
+	checker: qt.JSONEquals,
+	got:     `{"First": 47.11}`,
+	args: []interface{}{
+		&OuterJSON{
+			First: 47.11,
+		},
+	},
+	expectedNegateFailure: tilde2bq(`
+error:
+  unexpected success
+got:
+  ~{"First": 47.11}~
+want:
+  &quicktest_test.OuterJSON{
+      First:  47.11,
+      Second: nil,
+  }
+`),
+}, {
+	about:   "JSONEquals nested",
+	checker: qt.JSONEquals,
+	got:     `{"First": 47.11, "Last": [{"First": "Hello", "Second": 42}]}`,
+	args: []interface{}{
+		&OuterJSON{
+			First: 47.11,
+			Second: []*InnerJSON{
+				{First: "Hello", Second: 42},
+			},
+		},
+	},
+	expectedNegateFailure: tilde2bq(`
+error:
+  unexpected success
+got:
+  ~{"First": 47.11, "Last": [{"First": "Hello", "Second": 42}]}~
+want:
+  &quicktest_test.OuterJSON{
+      First:  47.11,
+      Second: {
+          &quicktest_test.InnerJSON{
+              First:  "Hello",
+              Second: 42,
+              Third:  {},
+          },
+      },
+  }
+`),
+}, {
+	about:   "JSONEquals nested with newline",
+	checker: qt.JSONEquals,
+	got: `{"First": 47.11, "Last": [{"First": "Hello", "Second": 42},
+			{"First": "World", "Third": {"F": false}}]}`,
+	args: []interface{}{
+		&OuterJSON{
+			First: 47.11,
+			Second: []*InnerJSON{
+				{First: "Hello", Second: 42},
+				{First: "World", Third: map[string]bool{
+					"F": false,
+				}},
+			},
+		},
+	},
+	expectedNegateFailure: `
+error:
+  unexpected success
+got:
+  "{\"First\": 47.11, \"Last\": [{\"First\": \"Hello\", \"Second\": 42},\n\t\t\t{\"First\": \"World\", \"Third\": {\"F\": false}}]}"
+want:
+  &quicktest_test.OuterJSON{
+      First:  47.11,
+      Second: {
+          &quicktest_test.InnerJSON{
+              First:  "Hello",
+              Second: 42,
+              Third:  {},
+          },
+          &quicktest_test.InnerJSON{
+              First:  "World",
+              Second: 0,
+              Third:  {"F":false},
+          },
+      },
+  }
+`,
+}, {
+	about:   "JSONEquals extra field",
+	checker: qt.JSONEquals,
+	got:     `{"NotThere": 1}`,
+	args: []interface{}{
+		&OuterJSON{
+			First: 2,
+		},
+	},
+	expectedCheckFailure: fmt.Sprintf(`
+error:
+  values are not deep equal
+diff (-got +want):
```

Comment:
I think we need to add JSONEquals to the checkers supported by export_test.WithVerbosity, otherwise this test would fail when running with -v.

Reply:
Maybe we should test both without and with -v in CI.

Reply:
good point. done.